lib/ansible/plugins/connection/grpc.py

```diff
@@ -0,0 +1,272 @@
+            if grpc:
+                self._sub_plugin = {'type': 'grpc', 'name': self._network_os, 'obj': grpc}
+                self.queue_message('log', 'loaded gRPC plugin for network_os %s' % self._network_os)
+                self.queue_message('log', 'network_os is set to %s' % self._network_os)
+            else:
+                raise AnsibleConnectionFailure('unable to load API plugin for network_os %s' % self._network_os)
+        else:
+            self._sub_plugin['type'] = 'external'
+            self.queue_message('warning', 'Unable to automatically determine gRPC implementation type.'
+                                          ' Please manually configure ansible_network_os value for this host')
+
+    def _connect(self):
+        """
+        Create GRPC connection to target host
+        :return: None
+        """
+        if not HAS_GRPC:
+            raise AnsibleError(
+                "grpcio is required to use the grpc connection type. Please run 'pip install grpcio'"
+            )
+
+        host = self.get_option('host')
+        if self.connected:
+            self.queue_message('log', 'gRPC connection to host %s already exist' % host)
+            return
+
+        port = self.get_option('port')
+        self._target = host if port is None else '%s:%d' % (host, port)
+        self._timeout = self.get_option('persistent_command_timeout')
+        self._login_credentials = [('username', self.get_option('remote_user')), ('password', self.get_option('password'))]
+        ssl_target_name_override = self.get_option('ssl_target_name_override')
+        if ssl_target_name_override:
+            self._channel_options = [('grpc.ssl_target_name_override', ssl_target_name_override), ]
+        else:
+            self._channel_options = None
+
+        certs = {}
+        private_key_file = self.get_option('private_key_file')
+        root_certificates_file = self.get_option('root_certificates_file')
+        certificate_chain_file = self.get_option('certificate_chain_file')
+
+        try:
+            if root_certificates_file:
+                with open(root_certificates_file, 'rb') as f:
+                    certs['root_certificates'] = f.read()
+            if private_key_file:
+                with open(private_key_file, 'rb') as f:
+                    certs['private_key'] = f.read()
+            if certificate_chain_file:
+                with open(certificate_chain_file, 'rb') as f:
+                    certs['certificate_chain'] = f.read()
+        except Exception as e:
+            raise AnsibleConnectionFailure('Failed to read certificate keys: %s' % e)
+
+        if certs:
+            creds = ssl_channel_credentials(**certs)
+            channel = secure_channel(self._target, creds, options=self._channel_options)
+        else:
+            channel = insecure_channel(self._target, options=self._channel_options)
+        self._channel = implementations.Channel(channel)
```

Comment:
```suggestion
        self.queue_message('vvv', "ESTABLISH GRPC CONNECTION FOR USER: %s on PORT %s TO %s" %
                                 (self.get_option('remote_user'), port, host))
        self._channel = implementations.Channel(channel)
```
This to be consistent with other connection plugins (netconf, paramiko_ssh, etc.)